Cython/Utility/AsyncGen.c

```diff
@@ -2,6 +2,26 @@
 // Try to keep it in sync by doing this from time to time:
 //    sed -e 's|__pyx_||ig'  Cython/Utility/AsyncGen.c | diff -udw - cpython/Objects/genobject.c | less
 
+//////////////////// AsyncGenerator.module_state_decls ////////////////////
+
+PyTypeObject *__pyx__PyAsyncGenWrappedValueType;
+PyTypeObject *__pyx__PyAsyncGenASendType;
+PyTypeObject *__pyx__PyAsyncGenAThrowType;
+PyTypeObject *__pyx_AsyncGenType;
+
+// Freelists boost performance 6-10%; they also reduce memory
+// fragmentation, as _PyAsyncGenWrappedValue and PyAsyncGenASend
+// are short-living objects that are instantiated for every
+// __anext__ call.
+
+#if CYTHON_USE_FREELISTS
+struct __pyx__PyAsyncGenWrappedValue *__Pyx_ag_value_freelist[_PyAsyncGen_MAXFREELIST];
+int __Pyx_ag_value_freelist_free;
+
+struct __pyx_PyAsyncGenASend *__Pyx_ag_asend_freelist[_PyAsyncGen_MAXFREELIST];
+int __Pyx_ag_asend_freelist_free;
+#endif
```

Comment:
We are sharing the types through the ABI module, right? Is the freelist feature flag part of the ABI module's name? It changes the implementation and can at least lead to unexpected behaviour if we mix modules compiled with and without freelists.

Reply:
Yeah I think you're right and it should be part of the ABI (and also `USE_TP_FINALIZE` too...).

It'd be a bit unusual to mix them since it's mostly just the free threading build that has it turned off I think. But there's no reason not to.